Library/include/CSP/Common/Array.h

```diff
@@ -74,7 +74,15 @@ template <typename T> class CSP_API Array
 		if (Buffer != nullptr && Size > 0)
 		{
 			AllocArray(Size);
-			memcpy(ObjectArray, Buffer, Size * sizeof(T));
+
+			// Perform the copy per-element via the assignment operator, rather than memcpy.
+			// This is to avoid introducing inadvertent shared memory ownership between elements in different arrays.
+			// We do not, for example, want strings in an array to hold the same pointers to
+			// memory as another array containing strings that it was constructed from.
+			for (size_t i = 0; i < Size; i++)
+			{
+				ObjectArray[i] = Buffer[i];
+			}
```

Comment:
The purpose of this constructor was to be able to "move" a raw C-style array into a `csp::common::Array`. The raw array would then be freed by the caller.

Reply:
I don't remember the specific instance that highlighted the issue here, but the issue is that if the memory being copied describes a set of non-PoD objects which hold pointers to memory that they control the lifetime over, then this constructor was leading to inadvertently shared memory lifetime management (because the pointer addresses would be directly copied to the new array's objects).

So then if one array was destroyed and the objects it described had their destructors called, and had their memory freed, then the other array's objects would be pointing at memory that was no longer valid.

To address this, this change replaces the memcpy with a per-element assignment, which makes use of the object's  assignment operator instead (making it the object's concern to safely handle memory management & object lifetimes).

Reply:
Since this change isn't directly related to the overall goals of the PR, I've restored it to using memcpy. This change is no longer part of the PR, so resolving the conversation.